// @baggie/core/lib/commonjs/color/_components/ColorViewer/ColorViewer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ColorViewer = void 0;
var _react = _interopRequireWildcard(require("react"));
var _isHexColor = require("../../check/isHexColor/isHexColor");
var _ColorViewerModule = _interopRequireDefault(require("./ColorViewer.module.scss"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const ColorViewer = _ref => {
  let {
    color,
    splitColor
  } = _ref;
  const isSplit = (0, _react.useMemo)(() => typeof splitColor === "string", [splitColor]);
  const isValidA = (0, _react.useMemo)(() => color ? (0, _isHexColor.isHexColor)(color) : false, [color]);
  const isValidB = (0, _react.useMemo)(() => isSplit ? (0, _isHexColor.isHexColor)(splitColor) : false, [isSplit, splitColor]);
  return /*#__PURE__*/_react.default.createElement("div", {
    className: _ColorViewerModule.default.wrapper
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: _ColorViewerModule.default.insideWrapper
  }, isValidA && (!isSplit || isValidB) || /*#__PURE__*/_react.default.createElement("svg", {
    height: "100px",
    width: "100px",
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 31 31",
    className: _ColorViewerModule.default.crossSvg
  }, /*#__PURE__*/_react.default.createElement("g", {
    transform: "rotate(45 15.5 15.5)",
    fill: "#900"
  }, /*#__PURE__*/_react.default.createElement("rect", {
    x: "15",
    y: "5",
    width: "1",
    height: "21"
  }), /*#__PURE__*/_react.default.createElement("rect", {
    x: "5",
    y: "15",
    width: "21",
    height: "1"
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: _ColorViewerModule.default.colorA,
    style: {
      border: `50px solid ${isSplit || !isValidA ? "transparent" : color}`,
      borderTopColor: isValidA ? color : "transparent",
      borderLeftColor: isValidA ? color : "transparent"
    }
  }), isSplit ? /*#__PURE__*/_react.default.createElement("div", {
    className: _ColorViewerModule.default.colorB,
    style: {
      borderRightColor: isValidB ? splitColor : "transparent",
      borderBottomColor: isValidB ? splitColor : "transparent"
    }
  }) : null), isSplit || /*#__PURE__*/_react.default.createElement("div", {
    className: _ColorViewerModule.default.text
  }, color));
};
exports.ColorViewer = ColorViewer;
