// @baggie/core/lib/commonjs/string/manipulation/stripTagsFromString/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Example = void 0;
var _react = _interopRequireWildcard(require("react"));
var _blocks = require("@storybook/blocks");
var _react2 = require("@baggie/react");
var _stripTagsFromString = require("./stripTagsFromString");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Example = props => {
  const [string, setString] = (0, _react.useState)(props.string);
  const [replaceWith, setReplaceWith] = (0, _react.useState)(props.replaceWith || " ");
  const [trim, setTrim] = (0, _react.useState)(true);
  const output = (0, _react.useMemo)(() => (0, _stripTagsFromString.stripTagsFromString)(string, replaceWith, trim), [string, replaceWith, trim]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      flexDirection: "column",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "String",
    labelForId: "string-input"
  }, /*#__PURE__*/_react.default.createElement("textarea", {
    id: "string-input",
    value: string,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setString(target.value);
    },
    style: {
      width: "100%",
      maxWidth: "360px"
    },
    spellCheck: false,
    rows: 3
  })), /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Replace with",
    labelForId: "replace-with-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "replace-with-input",
    value: replaceWith,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setReplaceWith(target.value);
    }
  })), /*#__PURE__*/_react.default.createElement("label", null, /*#__PURE__*/_react.default.createElement("input", {
    type: "checkbox",
    checked: trim,
    onChange: () => setTrim(current => !current)
  }), " ", "Trim"))), /*#__PURE__*/_react.default.createElement(_blocks.Source, {
    dark: true,
    code: `
import { stripTagsFromString } from "@baggie/core";

const string = \`${string.replace(/`/g, "\\`")}\`;
${replaceWith === " " ? "" : `const replaceWith = "${replaceWith}";\n`}
const output = stripTagsFromString(input${replaceWith === " " ? trim ? "" : ", undefined" : ", replaceWith"}${trim ? "" : ", false"});
/*
output = ${JSON.stringify(output, null, 4)}
*/
`
  }));
};
exports.Example = Example;
