// @baggie/core/lib/commonjs/color/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ColorComparison", {
  enumerable: true,
  get: function () {
    return _colorComparison.ColorComparison;
  }
});
Object.defineProperty(exports, "convertColorNameToHex", {
  enumerable: true,
  get: function () {
    return _convertColorNameToHex.convertColorNameToHex;
  }
});
Object.defineProperty(exports, "convertHslToRgb", {
  enumerable: true,
  get: function () {
    return _convertHslToRgb.convertHslToRgb;
  }
});
Object.defineProperty(exports, "convertRgbToHex", {
  enumerable: true,
  get: function () {
    return _convertRgbToHex.convertRgbToHex;
  }
});
Object.defineProperty(exports, "convertRgbToHsl", {
  enumerable: true,
  get: function () {
    return _convertRgbToHsl.convertRgbToHsl;
  }
});
Object.defineProperty(exports, "convertRgbToLab", {
  enumerable: true,
  get: function () {
    return _convertRgbToLab.convertRgbToLab;
  }
});
Object.defineProperty(exports, "convertStringToRgb", {
  enumerable: true,
  get: function () {
    return _convertStringToRgb.convertStringToRgb;
  }
});
Object.defineProperty(exports, "getColorBrightness", {
  enumerable: true,
  get: function () {
    return _getColorBrightness.getColorBrightness;
  }
});
Object.defineProperty(exports, "getColorDarkness", {
  enumerable: true,
  get: function () {
    return _getColorDarkness.getColorDarkness;
  }
});
Object.defineProperty(exports, "getColorSchemeAnalogous", {
  enumerable: true,
  get: function () {
    return _getColorSchemeAnalogous.getColorSchemeAnalogous;
  }
});
Object.defineProperty(exports, "getColorSchemeComplementary", {
  enumerable: true,
  get: function () {
    return _getColorSchemeComplementary.getColorSchemeComplementary;
  }
});
Object.defineProperty(exports, "getColorSchemeSplitComplementary", {
  enumerable: true,
  get: function () {
    return _getColorSchemeSplitComplementary.getColorSchemeSplitComplementary;
  }
});
Object.defineProperty(exports, "getColorSchemeTetradic", {
  enumerable: true,
  get: function () {
    return _getColorSchemeTetradic.getColorSchemeTetradic;
  }
});
Object.defineProperty(exports, "getColorSchemeTriadic", {
  enumerable: true,
  get: function () {
    return _getColorSchemeTriadic.getColorSchemeTriadic;
  }
});
Object.defineProperty(exports, "isColorBrighterThan", {
  enumerable: true,
  get: function () {
    return _isColorBrighterThan.isColorBrighterThan;
  }
});
Object.defineProperty(exports, "isColorDarkerThan", {
  enumerable: true,
  get: function () {
    return _isColorDarkerThan.isColorDarkerThan;
  }
});
Object.defineProperty(exports, "isHexColor", {
  enumerable: true,
  get: function () {
    return _isHexColor.isHexColor;
  }
});
Object.defineProperty(exports, "mixColors", {
  enumerable: true,
  get: function () {
    return _mixColors.mixColors;
  }
});
Object.defineProperty(exports, "multiplyColors", {
  enumerable: true,
  get: function () {
    return _multiplyColors.multiplyColors;
  }
});
Object.defineProperty(exports, "normalizeHexColor", {
  enumerable: true,
  get: function () {
    return _normalizeHexColor.normalizeHexColor;
  }
});
Object.defineProperty(exports, "screenColors", {
  enumerable: true,
  get: function () {
    return _screenColors.screenColors;
  }
});
var _getColorBrightness = require("./brightness/getColorBrightness/getColorBrightness");
var _getColorDarkness = require("./brightness/getColorDarkness/getColorDarkness");
var _isColorBrighterThan = require("./brightness/isColorBrighterThan/isColorBrighterThan");
var _isColorDarkerThan = require("./brightness/isColorDarkerThan/isColorDarkerThan");
var _isHexColor = require("./check/isHexColor/isHexColor");
var _getColorSchemeAnalogous = require("./colorSchemes/getColorSchemeAnalogous/getColorSchemeAnalogous");
var _getColorSchemeComplementary = require("./colorSchemes/getColorSchemeComplementary/getColorSchemeComplementary");
var _getColorSchemeSplitComplementary = require("./colorSchemes/getColorSchemeSplitComplementary/getColorSchemeSplitComplementary");
var _getColorSchemeTetradic = require("./colorSchemes/getColorSchemeTetradic/getColorSchemeTetradic");
var _getColorSchemeTriadic = require("./colorSchemes/getColorSchemeTriadic/getColorSchemeTriadic");
var _colorComparison = require("./comparison/colorComparison");
var _convertColorNameToHex = require("./convert/convertColorNameToHex/convertColorNameToHex");
var _convertStringToRgb = require("./convert/convertStringToRgb/convertStringToRgb");
var _convertHslToRgb = require("./convert/convertHslToRgb/convertHslToRgb");
var _convertRgbToHex = require("./convert/convertRgbToHex/convertRgbToHex");
var _convertRgbToHsl = require("./convert/convertRgbToHsl/convertRgbToHsl");
var _convertRgbToLab = require("./convert/convertRgbToLab/convertRgbToLab");
var _normalizeHexColor = require("./convert/normalizeHexColor/normalizeHexColor");
var _mixColors = require("./manipulate/mixColors/mixColors");
var _multiplyColors = require("./manipulate/multiplyColors/multiplyColors");
var _screenColors = require("./manipulate/screenColors/screenColors");
