// @baggie/core/lib/commonjs/math/time/getHalfLifeDecay/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Example = void 0;
var _react = _interopRequireWildcard(require("react"));
var _react2 = require("@baggie/react");
var _blocks = require("@storybook/blocks");
var _getHalfLifeDecay = require("./getHalfLifeDecay");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Example = props => {
  const [initialValue, setInitialValue] = (0, _react.useState)(props.initialValue);
  const [timePassed, setTimePassed] = (0, _react.useState)(props.timePassed);
  const [halfLife, setHalfLife] = (0, _react.useState)(props.halfLife);
  const reducedValue = (0, _react.useMemo)(() => (0, _getHalfLifeDecay.getHalfLifeDecay)(initialValue, timePassed || 0, halfLife || 0), [initialValue, timePassed, halfLife]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Initial value",
    labelForId: "value-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "value-input",
    type: "number",
    value: initialValue,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setInitialValue(target.valueAsNumber);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Time passed",
    labelForId: "time-passed-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "time-passed-input",
    type: "number",
    value: timePassed,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setTimePassed(target.valueAsNumber);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Half life",
    labelForId: "half-life-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "half-life-input",
    type: "number",
    value: halfLife,
    onChange: _ref3 => {
      let {
        target
      } = _ref3;
      return setHalfLife(target.valueAsNumber);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Reduced value",
    labelForId: undefined
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, isNaN(reducedValue) ? "Error" : reducedValue))), /*#__PURE__*/_react.default.createElement(_blocks.Source, {
    dark: true,
    code: `
import { getHalfLifeDecay } from "@baggie/core";

const initialValue = ${initialValue || 0};
const timePassed = ${timePassed || 0};
const halfLife = ${halfLife || 0};

const reducedValue = getHalfLifeDecay(initialValue, timePassed, halfLife);
// reducedValue = ${reducedValue};
`
  }));
};
exports.Example = Example;
