// @baggie/core/lib/commonjs/math/percentage/getPercentageBetween/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Example = void 0;
var _react = _interopRequireWildcard(require("react"));
var _blocks = require("@storybook/blocks");
var _react2 = require("@baggie/react");
var _getPercentageBetween = require("./getPercentageBetween");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Example = props => {
  const [from, setFrom] = (0, _react.useState)(props.from);
  const [to, setTo] = (0, _react.useState)(props.to);
  const [position, setPosition] = (0, _react.useState)(props.position);
  const result = (0, _react.useMemo)(() => (0, _getPercentageBetween.getPercentageBetween)(position, {
    from,
    to
  }), [from, to, position]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Position",
    labelForId: "position-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "position-input",
    type: "number",
    value: position,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setPosition(target.valueAsNumber || 0);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "From",
    labelForId: "from-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "from-input",
    type: "number",
    value: from,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setFrom(target.valueAsNumber || 0);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "To",
    labelForId: "to-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "to-input",
    type: "number",
    value: to,
    onChange: _ref3 => {
      let {
        target
      } = _ref3;
      return setTo(target.valueAsNumber || 0);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Percentage",
    labelForId: undefined
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, result))), /*#__PURE__*/_react.default.createElement(_blocks.Source, {
    dark: true,
    code: `
import { getPercentageBetween } from "@baggie/core";

const percentage = getPercentageBetween(${position}, { from: ${from}, to: ${to} });
// percentage = ${result}
`
  }));
};
exports.Example = Example;
