// @baggie/core/lib/module/color/brightness/getColorDarkness/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { InputWrapper } from "@baggie/react";
import { ColorInput } from "../../_components/ColorInput/ColorInput";
import { getColorDarkness } from "./getColorDarkness";
export const Example = props => {
  const [color, setColor] = useState(props.color);
  const darkness = useMemo(() => getColorDarkness(color), [color]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(ColorInput, {
    label: "Color",
    value: color,
    onChange: setColor
  }), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Darkness",
    labelForId: undefined
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, darkness))), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { getColorDarkness } from "@baggie/core";

const color = "${color}";

const darkness = getColorDarkness(color);
// darkness = ${darkness}
`
  }));
};
