// @baggie/core/lib/module/math/percentage/getPercentageIncrease/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { InputWrapper } from "@baggie/react";
import { getPercentageIncrease } from "./getPercentageIncrease";
export const Example = props => {
  const [from, setFrom] = useState(props.from);
  const [to, setTo] = useState(props.to);
  const [percentageAsDecimal, setPercentageAsDecimal] = useState(true);
  const result = useMemo(() => getPercentageIncrease(from, to, percentageAsDecimal), [from, to, percentageAsDecimal]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(InputWrapper, {
    label: "From",
    labelForId: "from-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "from-input",
    type: "number",
    value: from,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setFrom(target.valueAsNumber || 0);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "To",
    labelForId: "to-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "to-input",
    type: "number",
    value: to,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setTo(target.valueAsNumber || 0);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Percentage",
    labelForId: undefined
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, result))), /*#__PURE__*/React.createElement("label", null, /*#__PURE__*/React.createElement("input", {
    type: "checkbox",
    checked: percentageAsDecimal,
    onChange: () => setPercentageAsDecimal(current => !current)
  }), " ", "Percentage as decimal"), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { setPercentageAsDecimal } from "@baggie/core";

const from = ${from};
const to = ${to};

const percentage = setPercentageAsDecimal(from, to${percentageAsDecimal ? "" : ", false"});
// percentage = ${result}${percentageAsDecimal ? ` (${result * 100}%)` : ""}
`
  }));
};
