// @baggie/core/lib/commonjs/math/geometry/rectangle/getRectanglePerimeter/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Example = void 0;
var _react = _interopRequireWildcard(require("react"));
var _react2 = require("@baggie/react");
var _blocks = require("@storybook/blocks");
var _getRectanglePerimeter = require("./getRectanglePerimeter");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Example = props => {
  const [x, setX] = (0, _react.useState)(props.x);
  const [y, setY] = (0, _react.useState)(props.y);
  const perimeter = (0, _react.useMemo)(() => (0, _getRectanglePerimeter.getRectanglePerimeter)(x, y), [x, y]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "X",
    labelForId: "x-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "x-input",
    type: "number",
    value: x,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setX(target.valueAsNumber);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Y",
    labelForId: "x-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "y-input",
    type: "number",
    value: y,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setY(target.valueAsNumber);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Perimeter",
    labelForId: undefined
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, isNaN(perimeter) ? "Error" : perimeter))), /*#__PURE__*/_react.default.createElement("div", {
    style: {
      aspectRatio: `${Math.abs(x)} / ${Math.abs(y)}`,
      background: "#FC0",
      width: Math.abs(x) >= Math.abs(y) ? "150px" : `${Math.abs(x) / Math.abs(y) * 150}px`,
      height: Math.abs(x) <= Math.abs(y) ? "150px" : `${Math.abs(y) / Math.abs(x) * 150}px`,
      marginTop: "20px",
      position: "relative",
      display: "inline-block",
      pointerEvents: "none",
      userSelect: "none",
      transition: "all .4s ease-out"
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      position: "absolute",
      inset: "-4px",
      border: "2px dashed #900"
    }
  })), /*#__PURE__*/_react.default.createElement(_blocks.Source, {
    dark: true,
    code: `
import { getRectanglePerimeter } from "@baggie/core";

const x = ${x || 0};
const y = ${y || 0};

const perimeter = getRectanglePerimeter(x, y);
// perimeter = ${perimeter}
`
  }));
};
exports.Example = Example;
