// @baggie/core/lib/commonjs/math/range/isInRange/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Example = void 0;
var _react = _interopRequireWildcard(require("react"));
var _blocks = require("@storybook/blocks");
var _react2 = require("@baggie/react");
var _isInRange = require("./isInRange");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Example = props => {
  const [input, setInput] = (0, _react.useState)(props.number);
  const [ranges, setRanges] = (0, _react.useState)(props.ranges);
  const {
    realRanges,
    result
  } = (0, _react.useMemo)(() => {
    const realRanges = ranges.replace(/][^[{]+/g, "]\n").replace(/}[^[{]+/g, "}\n").split("\n").map(item => {
      try {
        return JSON.parse(item.replace(/["`']/g, "").replace(/([a-z]+)/g, `"$1"`));
      } catch (err) {
        return undefined;
      }
    }).filter(x => x && (Array.isArray(x) && x.length === 2 || typeof x === "object" && typeof x.from === "number" && typeof x.to === "number"));
    const result = (0, _isInRange.isInRange)(input, realRanges);
    return {
      realRanges,
      result
    };
  }, [input, ranges]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Number",
    labelForId: "number-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "number-input",
    type: "number",
    value: input,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setInput(target.valueAsNumber || 0);
    },
    autoComplete: "off"
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Ranges",
    labelForId: "ranges-input"
  }, /*#__PURE__*/_react.default.createElement("textarea", {
    id: "ranges-input",
    value: ranges,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setRanges(target.value);
    },
    style: {
      width: "100%",
      maxWidth: "360px"
    },
    autoComplete: "off",
    rows: 3
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Result",
    labelForId: undefined
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, result ? "true" : "false"))), /*#__PURE__*/_react.default.createElement(_blocks.Source, {
    dark: true,
    code: `
import { isInRange } from "@baggie/core";

const range${realRanges.length === 1 ? "" : "s"} = ${realRanges.length === 1 ? JSON.stringify(realRanges[0], null, 4) : JSON.stringify(realRanges, null, 4)};

const result = isInRange(${input}, range${realRanges.length === 1 ? "" : "s"});
// result = ${result ? "true" : "false"}
`
  }));
};
exports.Example = Example;
