// @baggie/core/lib/commonjs/detection/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "DetectInputMethod", {
  enumerable: true,
  get: function () {
    return _detectInputMethod.DetectInputMethod;
  }
});
Object.defineProperty(exports, "isBrowser", {
  enumerable: true,
  get: function () {
    return _isBrowser.isBrowser;
  }
});
Object.defineProperty(exports, "isJsDom", {
  enumerable: true,
  get: function () {
    return _isJsDom.isJsDom;
  }
});
Object.defineProperty(exports, "isNodeJs", {
  enumerable: true,
  get: function () {
    return _isNodeJs.isNodeJs;
  }
});
Object.defineProperty(exports, "isReactNative", {
  enumerable: true,
  get: function () {
    return _isReactNative.isReactNative;
  }
});
Object.defineProperty(exports, "isWebWorker", {
  enumerable: true,
  get: function () {
    return _isWebWorker.isWebWorker;
  }
});
Object.defineProperty(exports, "orientationIsLandscape", {
  enumerable: true,
  get: function () {
    return _orientationIsLandscape.orientationIsLandscape;
  }
});
Object.defineProperty(exports, "orientationIsPortrait", {
  enumerable: true,
  get: function () {
    return _orientationIsPortrait.orientationIsPortrait;
  }
});
Object.defineProperty(exports, "prefersDarkColorScheme", {
  enumerable: true,
  get: function () {
    return _prefersDarkColorScheme.prefersDarkColorScheme;
  }
});
Object.defineProperty(exports, "prefersReducedMotion", {
  enumerable: true,
  get: function () {
    return _prefersReducedMotion.prefersReducedMotion;
  }
});
var _isBrowser = require("./environment/isBrowser");
var _isJsDom = require("./environment/isJsDom");
var _isNodeJs = require("./environment/isNodeJs");
var _isReactNative = require("./environment/isReactNative");
var _isWebWorker = require("./environment/isWebWorker");
var _detectInputMethod = require("./input/detectInputMethod");
var _orientationIsLandscape = require("./mediaFeatures/orientationIsLandscape");
var _orientationIsPortrait = require("./mediaFeatures/orientationIsPortrait");
var _prefersDarkColorScheme = require("./mediaFeatures/prefersDarkColorScheme");
var _prefersReducedMotion = require("./mediaFeatures/prefersReducedMotion");
