// @baggie/core/lib/module/color/manipulate/multiplyColors/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { ColorViewer } from "../../_components/ColorViewer/ColorViewer";
import { ColorInput } from "../../_components/ColorInput/ColorInput";
import { multiplyColors } from "./multiplyColors";
export const Example = props => {
  const [colorA, setColorA] = useState(props.colorA);
  const [colorB, setColorB] = useState(props.colorB);
  const mixed = useMemo(() => multiplyColors(colorA, colorB), [colorA, colorB]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(ColorInput, {
    label: "Color A",
    value: colorA,
    onChange: setColorA
  }), /*#__PURE__*/React.createElement(ColorInput, {
    label: "Color B",
    value: colorB,
    onChange: setColorB
  })), /*#__PURE__*/React.createElement("div", {
    style: {
      marginTop: "25px"
    }
  }, /*#__PURE__*/React.createElement(ColorViewer, {
    color: mixed
  })), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { multiplyColors } from "@baggie/core";

const colorA = "${colorA}";
const colorB = "${colorB}";

const mixed = multiplyColors(colorA, colorB);
// mixed = ${mixed ? `"${mixed}"` : "undefined"}
`
  }));
};
