// @baggie/core/lib/commonjs/math/time/getDuration/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Example = void 0;
var _react = _interopRequireWildcard(require("react"));
var _blocks = require("@storybook/blocks");
var _react2 = require("@baggie/react");
var _getDuration = require("./getDuration");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Example = props => {
  const [speed, setSpeed] = (0, _react.useState)(props.speed);
  const [distance, setDistance] = (0, _react.useState)(props.distance);
  const duration = (0, _react.useMemo)(() => (0, _getDuration.getDuration)(speed, distance), [speed, distance]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Speed",
    labelForId: "speed-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "speed-input",
    type: "number",
    value: speed,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setSpeed(target.valueAsNumber);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Distance",
    labelForId: "distance-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "distance-input",
    type: "number",
    value: distance,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setDistance(target.valueAsNumber);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Duration",
    labelForId: undefined
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, isNaN(duration) ? "Error" : duration))), /*#__PURE__*/_react.default.createElement(_blocks.Source, {
    dark: true,
    code: `
import { getDuration } from "@baggie/core";

const speed = ${speed};
const distance = ${distance};

const duration = getDuration(speed, distance);
// duration = ${duration}
`
  }));
};
exports.Example = Example;
