// @baggie/core/lib/commonjs/math/time/getAge/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Example = void 0;
var _react = _interopRequireWildcard(require("react"));
var _blocks = require("@storybook/blocks");
var _react2 = require("@baggie/react");
var _getAge = require("./getAge");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Example = props => {
  const [fromTime, setFromTime] = (0, _react.useState)(props.birth);
  const [toTime, setToTime] = (0, _react.useState)(props.death);
  const fromString = new Date(fromTime).toISOString().replace(/\.[0-9]+Z$/, "");
  const toString = toTime ? new Date(toTime).toISOString().replace(/\.[0-9]+Z$/, "") : "";
  const age = (0, _react.useMemo)(() => (0, _getAge.getAge)(fromTime, toTime), [fromTime, toTime]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Birth",
    labelForId: "birth-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "birth-input",
    type: "datetime-local",
    value: fromString,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setFromTime(new Date(target.valueAsNumber));
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "To time",
    labelForId: "to-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "to-input",
    type: "datetime-local",
    value: toString,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setToTime(new Date(target.valueAsNumber));
    }
  }))), /*#__PURE__*/_react.default.createElement(_blocks.Source, {
    dark: true,
    code: `
import { getAge } from "@baggie/core";

const birth = "${fromTime.toDateString()}";
${toTime ? `const death = "${toTime === null || toTime === void 0 ? void 0 : toTime.toDateString()}";
` : ""}
const age = getAge(birth${toTime ? ", death" : ""});
// age = ${age}
`
  }));
};
exports.Example = Example;
