// @baggie/core/lib/module/string/manipulation/uppercaseFirstLetter/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { InputWrapper } from "@baggie/react";
import { uppercaseFirstLetter } from "./uppercaseFirstLetter";
export const Example = props => {
  const [input, setInput] = useState(props.defaultInput);
  const [lowercaseRest, setLowercaseRest] = useState(false);
  const output = useMemo(() => uppercaseFirstLetter(input, {
    lowercaseRest
  }), [input, lowercaseRest]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      flexDirection: "column",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Input",
    labelForId: "input-input"
  }, /*#__PURE__*/React.createElement("textarea", {
    id: "input-input",
    value: input,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setInput(target.value);
    },
    style: {
      width: "100%",
      maxWidth: "360px"
    },
    spellCheck: false,
    rows: 1
  })), /*#__PURE__*/React.createElement("label", null, /*#__PURE__*/React.createElement("input", {
    type: "checkbox",
    checked: lowercaseRest,
    onChange: () => setLowercaseRest(current => !current)
  }), "Lowercase rest")), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { uppercaseFirstLetter } from "@baggie/core";

const input = \`${input.replace(/`/g, "\\`")}\`;

const output = uppercaseFirstLetter(input${lowercaseRest ? ", { lowercaseRest: true }" : ""});
/*
output = \`${output}\`
*/
`
  }));
};
