// @baggie/core/lib/commonjs/color/convert/convertHslToRgb/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Example = void 0;
var _react = _interopRequireWildcard(require("react"));
var _blocks = require("@storybook/blocks");
var _react2 = require("@baggie/react");
var _ColorViewer = require("../../_components/ColorViewer/ColorViewer");
var _convertRgbToHex = require("../convertRgbToHex/convertRgbToHex");
var _convertHslToRgb = require("./convertHslToRgb");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Example = props => {
  const [hue, setHue] = (0, _react.useState)(props.hue);
  const [saturation, setSaturation] = (0, _react.useState)(props.saturation);
  const [lightness, setLightness] = (0, _react.useState)(props.lightness);
  const [alpha, setAlpha] = (0, _react.useState)(props.alpha);
  const {
    rgb,
    hex
  } = (0, _react.useMemo)(() => {
    const rgb = (0, _convertHslToRgb.convertHslToRgb)({
      hue,
      saturation,
      lightness,
      alpha
    });
    return {
      rgb,
      hex: (0, _convertRgbToHex.convertRgbToHex)(rgb)
    };
  }, [hue, saturation, lightness, alpha]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Hue",
    labelRight: hue,
    labelForId: "hue-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "hue-input",
    type: "range",
    min: 0,
    max: 360,
    step: 1,
    value: hue,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setHue(target.valueAsNumber);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Saturation",
    labelRight: saturation,
    labelForId: "saturation-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "saturation-input",
    type: "range",
    min: 0,
    max: 1,
    step: 0.01,
    value: saturation,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setSaturation(target.valueAsNumber);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Lightness",
    labelRight: lightness,
    labelForId: "lightness-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "lightness-input",
    type: "range",
    min: 0,
    max: 1,
    step: 0.01,
    value: lightness,
    onChange: _ref3 => {
      let {
        target
      } = _ref3;
      return setLightness(target.valueAsNumber);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Alpha",
    labelRight: alpha.toFixed(2),
    labelForId: "alpha-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "alpha-input",
    type: "range",
    min: 0,
    max: 1,
    step: 0.01,
    value: alpha,
    onChange: _ref4 => {
      let {
        target
      } = _ref4;
      return setAlpha(target.valueAsNumber);
    }
  }))), /*#__PURE__*/_react.default.createElement("div", {
    style: {
      marginTop: "25px",
      display: "flex",
      gap: "10px",
      alignItems: "flex-start"
    }
  }, /*#__PURE__*/_react.default.createElement(_ColorViewer.ColorViewer, {
    color: hex
  }), /*#__PURE__*/_react.default.createElement("table", {
    style: {
      margin: 0
    }
  }, /*#__PURE__*/_react.default.createElement("tbody", null, /*#__PURE__*/_react.default.createElement("tr", null, /*#__PURE__*/_react.default.createElement("td", null, /*#__PURE__*/_react.default.createElement("b", null, "Red:")), /*#__PURE__*/_react.default.createElement("td", null, rgb.red)), /*#__PURE__*/_react.default.createElement("tr", null, /*#__PURE__*/_react.default.createElement("td", null, /*#__PURE__*/_react.default.createElement("b", null, "Green:")), /*#__PURE__*/_react.default.createElement("td", null, rgb.green)), /*#__PURE__*/_react.default.createElement("tr", null, /*#__PURE__*/_react.default.createElement("td", null, /*#__PURE__*/_react.default.createElement("b", null, "Blue:")), /*#__PURE__*/_react.default.createElement("td", null, rgb.blue)), /*#__PURE__*/_react.default.createElement("tr", null, /*#__PURE__*/_react.default.createElement("td", null, /*#__PURE__*/_react.default.createElement("b", null, "Alpha:")), /*#__PURE__*/_react.default.createElement("td", null, rgb.alpha))))), /*#__PURE__*/_react.default.createElement(_blocks.Source, {
    dark: true,
    code: `
import { convertHslToRgb } from "@baggie/core";

const hsl = {
   hue: ${hue},
   saturation: ${saturation},
   lightness: ${lightness},
   alpha: ${alpha},
};

const rgb = convertHslToRgb(hsl);
/*
rgb = ${JSON.stringify(rgb, null, 4)}
*/
`
  }));
};
exports.Example = Example;
