// @baggie/core/lib/module/string/convert/getVimeoId/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { InputWrapper } from "@baggie/react";
import { getVimeoId } from "./getVimeoId";
export const Example = props => {
  const [input, setInput] = useState(props.input);
  const id = useMemo(() => getVimeoId(input), [input]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      flexWrap: "wrap",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Input",
    labelForId: "input-input"
  }, /*#__PURE__*/React.createElement("textarea", {
    id: "input-input",
    value: input,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setInput(target.value);
    },
    style: {
      width: "360px",
      maxWidth: "100%"
    },
    spellCheck: false,
    rows: 3
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "ID",
    labelForId: undefined
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, id || "undefined"))), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { getVimeoId } from "@baggie/core";

const input = \`${input}\`;

const id = getVimeoId(input);
// id = ${id ? `"${id}"` : "undefined"}
`
  }));
};
