// @baggie/core/lib/module/math/percentage/getPositionBetween/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { InputWrapper } from "@baggie/react";
import { getPositionBetween } from "./getPositionBetween";
export const Example = props => {
  const [from, setFrom] = useState(props.from);
  const [to, setTo] = useState(props.to);
  const [percentage, setPercentage] = useState(props.percentage);
  const result = useMemo(() => getPositionBetween(from, to, percentage), [from, to, percentage]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(InputWrapper, {
    label: "From",
    labelForId: "from-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "from-input",
    type: "number",
    value: from,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setFrom(target.valueAsNumber || 0);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "To",
    labelForId: "to-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "to-input",
    type: "number",
    value: to,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setTo(target.valueAsNumber || 0);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Percentage",
    labelForId: "percentage-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "percentage-input",
    type: "number",
    value: percentage,
    onChange: _ref3 => {
      let {
        target
      } = _ref3;
      return setPercentage(target.valueAsNumber || 0);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Position",
    labelForId: undefined
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, result))), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { getPositionBetween } from "@baggie/core";

const from = ${from};
const to = ${to};
const percentage = ${percentage};

const position = getPositionBetween(from, to, percentage);
// position = ${result}
`
  }));
};
