// colors162/index.js
const chalk = require(`chalk`)
var asdwasdjniuscnac = 1

async function GradRed(prefix, text, intensity){
    var zizi = 21*Math.sin(3* intensity)**2
    var r = Math.round(255 - (12*zizi))
    var g = 0*zizi 
    console.log(chalk.rgb(r, g, 0)(prefix), text)
}

// async function GradRed2(prefix, text, intensity){
//     r = 0 + (12*intensity)
//     g = 0*intensity 
//     console.log(chalk.rgb(r, g, 0)(prefix), text, r, intensity)
// }

async function GradBluePre(prefix, text, intensity){
    var zizi = 12*Math.sin(3* intensity)**2
    b = 255 - Math.round(12*zizi)
    g = 122 - Math.round(10*zizi)
    console.log(chalk.rgb(0, g, b)(prefix), text)
}

async function GradGreenPre(prefix, text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    r = 69 - Math.round(6*intensity)
    g = 255 - Math.round(12*intensity)
    console.log(chalk.rgb(r, g, 0)(prefix), text)
}

async function GradPinkPre(prefix, text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    r = 255 - Math.round(12*zizi)
    b = 187 - Math.round(16*zizi)
    console.log(chalk.rgb(r, 0, b)(prefix), text)
}

async function GradYellowPre(prefix, text, intensity){
    var zizi = 21*Math.sin(3* intensity)**2
    r = 255 - Math.round(12*zizi)
    g = 247 - Math.round(11*zizi)
    console.log(chalk.rgb(r, g, 0)(prefix), text)
}

async function GradLightbluePre(prefix, text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    g = 255 - Math.round(12*zizi)
    b = 238 - Math.round(20*zizi)
    console.log(chalk.rgb(0, g, b)(prefix), text)
}

async function GradOrangePre(prefix, text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    r = 255 - Math.round(12*zizi)
    g = 187 - Math.round(16*zizi)
    console.log(chalk.rgb(r, g, 0)(prefix), text)
}

async function GradPurplePre(prefix, text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    b = 255 - Math.round(12*zizi)
    r = 151 - Math.round(13*zizi)
    console.log(chalk.rgb(r, 0, b)(prefix), text)
}

async function GradPeachPre(prefix, text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    r = 255 - Math.round(12*zizi)
    g = 187 - Math.round(16*zizi)
    b = 126 - Math.round(4*zizi)
    console.log(chalk.rgb(r, g, b)(prefix), text)
}

async function GradTurquoisePre(prefix, text, intensity){
    var zizi = 10*Math.sin(3* intensity)**2
    r = 123 - Math.round(12*zizi)
    g = 213 - Math.round(16*zizi)
    b = 182 - Math.round(4*zizi)
    console.log(chalk.rgb(r, g, b)(prefix), text)
}

async function GradEmeraldPre(prefix, text, intensity){
    var zizi = 5*Math.sin(3* intensity)**2
    r = 62 - Math.round(12*zizi)
    g = 182 - Math.round(16*zizi)
    b = 88 - Math.round(4*zizi)
    console.log(chalk.rgb(r, g, b)(prefix), text)
}

const logRedPre = (prefix, text)=>{
    GradRed(prefix, text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}



const logBluePre = (prefix, text)=>{
    GradBlue(prefix, text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}



const logGreenPre = (prefix, text)=>{
    GradGreen(prefix, text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}



const logPinkPre = (prefix, text)=>{
    GradPink(prefix, text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}



const logLightBluePre = (prefix, text)=>{
    GradLightblue(prefix, text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}



const logYellowPre = (prefix, text)=>{
    GradYellow(prefix, text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}



const logOrangePre = (prefix, text)=>{
    GradOrange(prefix, text, asdwasdjniuscnac)
    nasdwasdjniuscnac++
}



const logPurplePre = (prefix, text)=>{
    GradPurple(prefix, text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}

const logPeachPre = (prefix, text)=>{
    GradPeach(prefix, text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}

const logTurquoisePre = (prefix, text)=>{
    GradTurquoise(prefix, text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}

const logEmeraldPre = (prefix, text)=>{
    GradEmerald(prefix, text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}

async function GradRed(text, intensity){
    var zizi = 21*Math.sin(3* intensity)**2
    var r = Math.round(255 - (12*zizi))
    var g = 0*zizi 
    console.log(chalk.rgb(r, g, 0)(text))
}

async function GradBlue(text, intensity){
    var zizi = 12*Math.sin(3* intensity)**2
    b = 255 - Math.round(12*zizi)
    g = 122 - Math.round(10*zizi)
    console.log(chalk.rgb(0, g, b)(text))
}

async function GradGreen( text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    r = 69 - Math.round(6*intensity)
    g = 255 - Math.round(12*intensity)
    console.log(chalk.rgb(r, g, 0)(text))
}

async function GradPink(text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    r = 255 - Math.round(12*zizi)
    b = 187 - Math.round(16*zizi)
    console.log(chalk.rgb(r, 0, b)(text))
}

async function GradYellow(ext, intensity){
    var zizi = 21*Math.sin(3* intensity)**2
    r = 255 - Math.round(12*zizi)
    g = 247 - Math.round(11*zizi)
    console.log(chalk.rgb(r, g, 0)(text))
}

async function GradLightblue(text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    g = 255 - Math.round(12*zizi)
    b = 238 - Math.round(20*zizi)
    console.log(chalk.rgb(0, g, b)(text))
}

async function GradOrange(text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    r = 255 - Math.round(12*zizi)
    g = 187 - Math.round(16*zizi)
    console.log(chalk.rgb(r, g, 0)(text))
}

async function GradPurple(text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    b = 255 - Math.round(12*zizi)
    r = 151 - Math.round(13*zizi)
    console.log(chalk.rgb(r, 0, b)(text))
}

async function GradPeach(text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    r = 255 - Math.round(12*zizi)
    g = 187 - Math.round(16*zizi)
    b = 126 - Math.round(4*zizi)
    console.log(chalk.rgb(r, g, b)(text))
}

async function GradTurquoise(text, intensity){
    var zizi = 10*Math.sin(3* intensity)**2
    r = 123 - Math.round(12*zizi)
    g = 213 - Math.round(16*zizi)
    b = 182 - Math.round(4*zizi)
    console.log(chalk.rgb(r, g, b)(text))
}

async function GradEmerald(text, intensity){
    var zizi = 5*Math.sin(3* intensity)**2
    r = 62 - Math.round(12*zizi)
    g = 182 - Math.round(16*zizi)
    b = 88 - Math.round(4*zizi)
    console.log(chalk.rgb(r, g, b)(text))
}

const logRed = (text)=>{
    GradRed(text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}



const logBlue = (text)=>{
    GradBlue(text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}



const logGreen = (text)=>{
    GradGreen(text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}



const logPink = (text)=>{
    GradPink(text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}



const logLightBlue = (text)=>{
    GradLightblue(text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}



const logYellow = (text)=>{
    GradYellow(text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}



const logOrange = (text)=>{
    GradOrange(text, asdwasdjniuscnac)
    nasdwasdjniuscnac++
}



const logPurple = (text)=>{
    GradPurple(text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}

const logPeach = (text)=>{
    GradPeach(text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}

const logTurquoise = (text)=>{
    GradTurquoise(text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}

const logEmerald = (text)=>{
    GradEmerald(text, asdwasdjniuscnac)
    asdwasdjniuscnac++
}

function Red1(text, intensity){
    var zizi = 21*Math.sin(3* intensity)**2
    var r = Math.round(255 - (12*zizi))
    var g = 0*zizi 
    return chalk.rgb(r, g, 0)(text)
}


function Blue1(text, intensity){
    var zizi = 12*Math.sin(3* intensity)**2
    b = 255 - Math.round(12*zizi)
    g = 122 - Math.round(10*zizi)
    return chalk.rgb(0, g, b)(text)
}

function Green1( text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    r = 69 - Math.round(6*intensity)
    g = 255 - Math.round(12*intensity)
    return chalk.rgb(r, g, 0)(text)
}

function Pink1(text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    r = 255 - Math.round(12*zizi)
    b = 187 - Math.round(16*zizi)
    return chalk.rgb(r, 0, b)(text)
}

function Yellow1(ext, intensity){
    var zizi = 21*Math.sin(3* intensity)**2
    r = 255 - Math.round(12*zizi)
    g = 247 - Math.round(11*zizi)
    return chalk.rgb(r, g, 0)(text)
}

function Lightblue1(text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    g = 255 - Math.round(12*zizi)
    b = 238 - Math.round(20*zizi)
    return chalk.rgb(0, g, b)(text)
}

function Orange1(text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    r = 255 - Math.round(12*zizi)
    g = 187 - Math.round(16*zizi)
    return chalk.rgb(r, g, 0)(text)
}

function Purple1(text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    b = 255 - Math.round(12*zizi)
    r = 151 - Math.round(13*zizi)
    return chalk.rgb(r, 0, b)(text)
}

function Peach1(text, intensity){
    var zizi = 11*Math.sin(3* intensity)**2
    r = 255 - Math.round(12*zizi)
    g = 187 - Math.round(16*zizi)
    b = 126 - Math.round(4*zizi)
    return chalk.rgb(r, g, b)(text)
}

function Turquoise1(text, intensity){
    var zizi = 10*Math.sin(3* intensity)**2
    r = 123 - Math.round(12*zizi)
    g = 213 - Math.round(16*zizi)
    b = 182 - Math.round(4*zizi)
    return chalk.rgb(r, g, b)(text)
}

function Emerald1(text, intensity){
    var zizi = 5*Math.sin(3* intensity)**2
    r = 62 - Math.round(12*zizi)
    g = 182 - Math.round(16*zizi)
    b = 88 - Math.round(4*zizi)
    return chalk.rgb(r,g,b)(text)
}

const Red = (text)=>{
    asdwasdjniuscnac++
    return Red1(text, asdwasdjniuscnac)
}



const Blue = (text)=>{
    asdwasdjniuscnac++
    return Blue1(text, asdwasdjniuscnac)
}



const Green = (text)=>{
    asdwasdjniuscnac++
    return Green1(text, asdwasdjniuscnac)
}



const Pink = (text)=>{
    asdwasdjniuscnac++
    return Pink1(text, asdwasdjniuscnac)
}



const LightBlue = (text)=>{
    asdwasdjniuscnac++
    return Lightblue1(text, asdwasdjniuscnac)
}



const Yellow = (text)=>{
    asdwasdjniuscnac++
    return Yellow1(text, asdwasdjniuscnac)
}



const Orange = (text)=>{
    asdwasdjniuscnac++
    return Orange1(text, asdwasdjniuscnac)
}



const Purple = (text)=>{
    asdwasdjniuscnac++
    return Purple1(text, asdwasdjniuscnac)
}

const Peach = (text)=>{
    asdwasdjniuscnac++
    return Peach1(text, asdwasdjniuscnac)
}

const Turquoise = (text)=>{
    asdwasdjniuscnac++
    return Turquoise1(text, asdwasdjniuscnac)
}

const Emerald = (text)=>{
    asdwasdjniuscnac++
    return Emerald1(text, asdwasdjniuscnac)
}

async function resetIntensity(){
    asdwasdjniuscnac = 1
}




module.exports = {
    logPurple,
    logOrange,
    logYellow,
    logLightBlue,
    logPink,
    logGreen,
    logBlue,
    logRed,
    logPeach,
    logTurquoise,
    logEmeraldPre,
    logPurplePre,
    logOrangePre,
    logYellowPre,
    logLightBluePre,
    logPinkPre,
    logGreenPre,
    logBluePre,
    logRedPre,
    logPeachPre,
    logTurquoisePre,
    logEmeraldPre,
    resetIntensity,
    asdwasdjniuscnac,
    Red,
    Blue,
    Green,
    Yellow,
    Purple,
    Orange,
    Pink,
    Peach,
    Turquoise,
    Emerald,
    LightBlue
 }
