// @baggie/core/lib/commonjs/color/check/isHexColor/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Example = void 0;
var _react = _interopRequireWildcard(require("react"));
var _blocks = require("@storybook/blocks");
var _react2 = require("@baggie/react");
var _ColorViewer = require("../../_components/ColorViewer/ColorViewer");
var _ColorInput = require("../../_components/ColorInput/ColorInput");
var _isHexColor = require("./isHexColor");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Example = props => {
  const [input, setInput] = (0, _react.useState)(props.defaultInput);
  const checkResult = (0, _react.useMemo)(() => (0, _isHexColor.isHexColor)(input), [input]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_ColorInput.ColorInput, {
    label: "Input",
    value: input,
    onChange: setInput
  }), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Is hex color",
    labelForId: undefined
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, checkResult ? "true" : "false"))), /*#__PURE__*/_react.default.createElement("div", {
    style: {
      marginTop: "25px"
    }
  }, /*#__PURE__*/_react.default.createElement(_ColorViewer.ColorViewer, {
    color: input
  })), /*#__PURE__*/_react.default.createElement(_blocks.Source, {
    dark: true,
    code: `
import { isHexColor } from "@baggie/core";

const input = "${input}";

const answer = isHexColor(input);
// answer = ${checkResult ? "true" : "false"}
`
  }));
};
exports.Example = Example;
