// @baggie/core/lib/module/color/check/isHexColor/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { InputWrapper } from "@baggie/react";
import { ColorViewer } from "../../_components/ColorViewer/ColorViewer";
import { ColorInput } from "../../_components/ColorInput/ColorInput";
import { isHexColor } from "./isHexColor";
export const Example = props => {
  const [input, setInput] = useState(props.defaultInput);
  const checkResult = useMemo(() => isHexColor(input), [input]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(ColorInput, {
    label: "Input",
    value: input,
    onChange: setInput
  }), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Is hex color",
    labelForId: undefined
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, checkResult ? "true" : "false"))), /*#__PURE__*/React.createElement("div", {
    style: {
      marginTop: "25px"
    }
  }, /*#__PURE__*/React.createElement(ColorViewer, {
    color: input
  })), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { isHexColor } from "@baggie/core";

const input = "${input}";

const answer = isHexColor(input);
// answer = ${checkResult ? "true" : "false"}
`
  }));
};
