// @baggie/core/lib/commonjs/color/brightness/isColorBrighterThan/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Example = void 0;
var _react = _interopRequireWildcard(require("react"));
var _blocks = require("@storybook/blocks");
var _react2 = require("@baggie/react");
var _ColorViewer = require("../../_components/ColorViewer/ColorViewer");
var _ColorInput = require("../../_components/ColorInput/ColorInput");
var _isColorBrighterThan = require("./isColorBrighterThan");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Example = props => {
  const [colorA, setColorA] = (0, _react.useState)(props.colorA);
  const [colorB, setColorB] = (0, _react.useState)(props.colorB);
  const isBrighter = (0, _react.useMemo)(() => (0, _isColorBrighterThan.isColorBrighterThan)(colorA, colorB), [colorA, colorB]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_ColorInput.ColorInput, {
    label: "Color A",
    value: colorA,
    onChange: setColorA
  }), /*#__PURE__*/_react.default.createElement(_ColorInput.ColorInput, {
    label: "Color B",
    value: colorB,
    onChange: setColorB
  }), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Is A brighter than B",
    labelForId: undefined
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, isBrighter ? "true" : "false"))), /*#__PURE__*/_react.default.createElement("div", {
    style: {
      marginTop: "25px"
    }
  }, /*#__PURE__*/_react.default.createElement(_ColorViewer.ColorViewer, {
    color: colorA,
    splitColor: colorB
  })), /*#__PURE__*/_react.default.createElement(_blocks.Source, {
    dark: true,
    code: `
import { isColorBrighterThan } from "@baggie/core";

const colorA = "${colorA}";
const colorB = "${colorB}";

const answer = isColorBrighterThan(colorA, colorB);
// answer = ${isBrighter ? "true" : "false"}
`
  }));
};
exports.Example = Example;
