// @baggie/core/lib/commonjs/math/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "addNumbers", {
  enumerable: true,
  get: function () {
    return _addNumbers.addNumbers;
  }
});
Object.defineProperty(exports, "ceilDecimals", {
  enumerable: true,
  get: function () {
    return _ceilDecimals.ceilDecimals;
  }
});
Object.defineProperty(exports, "ceilToMultipleOf", {
  enumerable: true,
  get: function () {
    return _ceilToMultipleOf.ceilToMultipleOf;
  }
});
Object.defineProperty(exports, "clamp", {
  enumerable: true,
  get: function () {
    return _clamp.clamp;
  }
});
Object.defineProperty(exports, "convertRangePosition", {
  enumerable: true,
  get: function () {
    return _convertRangePosition.convertRangePosition;
  }
});
Object.defineProperty(exports, "convertToCoordinates", {
  enumerable: true,
  get: function () {
    return _convertToCoordinates.convertToCoordinates;
  }
});
Object.defineProperty(exports, "countDecimals", {
  enumerable: true,
  get: function () {
    return _countDecimals.countDecimals;
  }
});
Object.defineProperty(exports, "degreesToRadians", {
  enumerable: true,
  get: function () {
    return _degreesToRadians.degreesToRadians;
  }
});
Object.defineProperty(exports, "floorDecimals", {
  enumerable: true,
  get: function () {
    return _floorDecimals.floorDecimals;
  }
});
Object.defineProperty(exports, "floorToMultipleOf", {
  enumerable: true,
  get: function () {
    return _floorToMultipleOf.floorToMultipleOf;
  }
});
Object.defineProperty(exports, "getAge", {
  enumerable: true,
  get: function () {
    return _getAge.getAge;
  }
});
Object.defineProperty(exports, "getAllDistances", {
  enumerable: true,
  get: function () {
    return _getAllDistances.getAllDistances;
  }
});
Object.defineProperty(exports, "getAntiMode", {
  enumerable: true,
  get: function () {
    return _getAntiMode.getAntiMode;
  }
});
Object.defineProperty(exports, "getBounds", {
  enumerable: true,
  get: function () {
    return _getBounds.getBounds;
  }
});
Object.defineProperty(exports, "getCircleArea", {
  enumerable: true,
  get: function () {
    return _getCircleArea.getCircleArea;
  }
});
Object.defineProperty(exports, "getCircleCircumference", {
  enumerable: true,
  get: function () {
    return _getCircleCircumference.getCircleCircumference;
  }
});
Object.defineProperty(exports, "getConeArea", {
  enumerable: true,
  get: function () {
    return _getConeArea.getConeArea;
  }
});
Object.defineProperty(exports, "getConeVolume", {
  enumerable: true,
  get: function () {
    return _getConeVolume.getConeVolume;
  }
});
Object.defineProperty(exports, "getCoordinatesAngle", {
  enumerable: true,
  get: function () {
    return _getCoordinatesAngle.getCoordinatesAngle;
  }
});
Object.defineProperty(exports, "getCuboidArea", {
  enumerable: true,
  get: function () {
    return _getCuboidArea.getCuboidArea;
  }
});
Object.defineProperty(exports, "getCuboidVolume", {
  enumerable: true,
  get: function () {
    return _getCuboidVolume.getCuboidVolume;
  }
});
Object.defineProperty(exports, "getCylinderArea", {
  enumerable: true,
  get: function () {
    return _getCylinderArea.getCylinderArea;
  }
});
Object.defineProperty(exports, "getCylinderVolume", {
  enumerable: true,
  get: function () {
    return _getCylinderVolume.getCylinderVolume;
  }
});
Object.defineProperty(exports, "getDistance", {
  enumerable: true,
  get: function () {
    return _getDistance.getDistance;
  }
});
Object.defineProperty(exports, "getDistanceOnEarth", {
  enumerable: true,
  get: function () {
    return _getDistanceOnEarth.getDistanceOnEarth;
  }
});
Object.defineProperty(exports, "getDistanceOnSphere", {
  enumerable: true,
  get: function () {
    return _getDistanceOnSphere.getDistanceOnSphere;
  }
});
Object.defineProperty(exports, "getDuration", {
  enumerable: true,
  get: function () {
    return _getDuration.getDuration;
  }
});
Object.defineProperty(exports, "getFrequency", {
  enumerable: true,
  get: function () {
    return _getFrequency.getFrequency;
  }
});
Object.defineProperty(exports, "getHalfLifeDecay", {
  enumerable: true,
  get: function () {
    return _getHalfLifeDecay.getHalfLifeDecay;
  }
});
Object.defineProperty(exports, "getMean", {
  enumerable: true,
  get: function () {
    return _getMean.getMean;
  }
});
Object.defineProperty(exports, "getMedian", {
  enumerable: true,
  get: function () {
    return _getMedian.getMedian;
  }
});
Object.defineProperty(exports, "getMode", {
  enumerable: true,
  get: function () {
    return _getMode.getMode;
  }
});
Object.defineProperty(exports, "getPercentage", {
  enumerable: true,
  get: function () {
    return _getPercentage.getPercentage;
  }
});
Object.defineProperty(exports, "getPercentageBetween", {
  enumerable: true,
  get: function () {
    return _getPercentageBetween.getPercentageBetween;
  }
});
Object.defineProperty(exports, "getPercentageIncrease", {
  enumerable: true,
  get: function () {
    return _getPercentageIncrease.getPercentageIncrease;
  }
});
Object.defineProperty(exports, "getPointBetween", {
  enumerable: true,
  get: function () {
    return _getPointBetween.getPointBetween;
  }
});
Object.defineProperty(exports, "getPointsOnCircle", {
  enumerable: true,
  get: function () {
    return _getPointsOnCircle.getPointsOnCircle;
  }
});
Object.defineProperty(exports, "getPositionBetween", {
  enumerable: true,
  get: function () {
    return _getPositionBetween.getPositionBetween;
  }
});
Object.defineProperty(exports, "getRange", {
  enumerable: true,
  get: function () {
    return _getRange.getRange;
  }
});
Object.defineProperty(exports, "getRectangleArea", {
  enumerable: true,
  get: function () {
    return _getRectangleArea.getRectangleArea;
  }
});
Object.defineProperty(exports, "getRectanglePerimeter", {
  enumerable: true,
  get: function () {
    return _getRectanglePerimeter.getRectanglePerimeter;
  }
});
Object.defineProperty(exports, "getSlope", {
  enumerable: true,
  get: function () {
    return _getSlope.getSlope;
  }
});
Object.defineProperty(exports, "getSphereArea", {
  enumerable: true,
  get: function () {
    return _getSphereArea.getSphereArea;
  }
});
Object.defineProperty(exports, "getSphereVolume", {
  enumerable: true,
  get: function () {
    return _getSphereVolume.getSphereVolume;
  }
});
Object.defineProperty(exports, "getTimeDifference", {
  enumerable: true,
  get: function () {
    return _getTimeDifference.getTimeDifference;
  }
});
Object.defineProperty(exports, "getTriangleArea", {
  enumerable: true,
  get: function () {
    return _getTriangleArea.getTriangleArea;
  }
});
Object.defineProperty(exports, "getTrianglePerimeter", {
  enumerable: true,
  get: function () {
    return _getTrianglePerimeter.getTrianglePerimeter;
  }
});
Object.defineProperty(exports, "isAngleInRange", {
  enumerable: true,
  get: function () {
    return _isAngleInRange.isAngleInRange;
  }
});
Object.defineProperty(exports, "isInRange", {
  enumerable: true,
  get: function () {
    return _isInRange.isInRange;
  }
});
Object.defineProperty(exports, "isInsideRadius", {
  enumerable: true,
  get: function () {
    return _isInsideRadius.isInsideRadius;
  }
});
Object.defineProperty(exports, "isNumberRound", {
  enumerable: true,
  get: function () {
    return _isNumberRound.isNumberRound;
  }
});
Object.defineProperty(exports, "keepInsideRadius", {
  enumerable: true,
  get: function () {
    return _keepInsideRadius.keepInsideRadius;
  }
});
Object.defineProperty(exports, "keepOutsideRange", {
  enumerable: true,
  get: function () {
    return _keepOutsideRange.keepOutsideRange;
  }
});
Object.defineProperty(exports, "mergeIntervals", {
  enumerable: true,
  get: function () {
    return _mergeIntervals.mergeIntervals;
  }
});
Object.defineProperty(exports, "nonNumberFallback", {
  enumerable: true,
  get: function () {
    return _nonNumberFallback.nonNumberFallback;
  }
});
Object.defineProperty(exports, "radiansToDegrees", {
  enumerable: true,
  get: function () {
    return _radiansToDegrees.radiansToDegrees;
  }
});
Object.defineProperty(exports, "roundDecimals", {
  enumerable: true,
  get: function () {
    return _roundDecimals.roundDecimals;
  }
});
Object.defineProperty(exports, "roundToMultipleOf", {
  enumerable: true,
  get: function () {
    return _roundToMultipleOf.roundToMultipleOf;
  }
});
Object.defineProperty(exports, "sortByClosestAngle", {
  enumerable: true,
  get: function () {
    return _sortByClosestAngle.sortByClosestAngle;
  }
});
Object.defineProperty(exports, "sortByDifference", {
  enumerable: true,
  get: function () {
    return _sortByDifference.sortByDifference;
  }
});
Object.defineProperty(exports, "truncDecimals", {
  enumerable: true,
  get: function () {
    return _truncDecimals.truncDecimals;
  }
});
Object.defineProperty(exports, "zeroIfNotNumber", {
  enumerable: true,
  get: function () {
    return _zeroIfNotNumber.zeroIfNotNumber;
  }
});
var _addNumbers = require("./arithmetic/addNumbers/addNumbers");
var _countDecimals = require("./arithmetic/countDecimals/countDecimals");
var _sortByDifference = require("./arithmetic/sortByDifference/sortByDifference");
var _getAntiMode = require("./average/getAntiMode/getAntiMode");
var _getFrequency = require("./average/getFrequency/getFrequency");
var _getMean = require("./average/getMean/getMean");
var _getMedian = require("./average/getMedian/getMedian");
var _getMode = require("./average/getMode/getMode");
var _convertToCoordinates = require("./coordinates/convertToCoordinates");
var _getCoordinatesAngle = require("./coordinates/angle/getCoordinatesAngle/getCoordinatesAngle");
var _getBounds = require("./coordinates/bounds/getBounds");
var _getAllDistances = require("./coordinates/distance/getAllDistances");
var _getDistance = require("./coordinates/distance/getDistance");
var _getDistanceOnEarth = require("./coordinates/distance/getDistanceOnEarth");
var _getDistanceOnSphere = require("./coordinates/distance/getDistanceOnSphere");
var _getPointBetween = require("./coordinates/distance/getPointBetween");
var _isInsideRadius = require("./coordinates/radius/isInsideRadius");
var _keepInsideRadius = require("./coordinates/radius/keepInsideRadius");
var _getSlope = require("./coordinates/slope/getSlope");
var _isAngleInRange = require("./geometry/angle/isAngleInRange");
var _sortByClosestAngle = require("./geometry/angle/sortByClosestAngle");
var _getCircleArea = require("./geometry/circle/getCircleArea/getCircleArea");
var _getCircleCircumference = require("./geometry/circle/getCircleCircumference/getCircleCircumference");
var _getPointsOnCircle = require("./geometry/circle/getPointsOnCircle/getPointsOnCircle");
var _getConeArea = require("./geometry/cone/getConeArea");
var _getConeVolume = require("./geometry/cone/getConeVolume");
var _getCuboidArea = require("./geometry/cuboid/getCuboidArea");
var _getCuboidVolume = require("./geometry/cuboid/getCuboidVolume");
var _getCylinderArea = require("./geometry/cylinder/getCylinderArea");
var _getCylinderVolume = require("./geometry/cylinder/getCylinderVolume");
var _getRectanglePerimeter = require("./geometry/rectangle/getRectanglePerimeter/getRectanglePerimeter");
var _getRectangleArea = require("./geometry/rectangle/getRectangleArea/getRectangleArea");
var _getSphereArea = require("./geometry/sphere/getSphereArea");
var _getSphereVolume = require("./geometry/sphere/getSphereVolume");
var _getTrianglePerimeter = require("./geometry/triangle/getTrianglePerimeter");
var _getTriangleArea = require("./geometry/triangle/getTriangleArea");
var _nonNumberFallback = require("./helpers/nonNumberFallback/nonNumberFallback");
var _zeroIfNotNumber = require("./helpers/zeroIfNotNumber/zeroIfNotNumber");
var _getPercentage = require("./percentage/getPercentage/getPercentage");
var _getPercentageBetween = require("./percentage/getPercentageBetween/getPercentageBetween");
var _getPercentageIncrease = require("./percentage/getPercentageIncrease/getPercentageIncrease");
var _getPositionBetween = require("./percentage/getPositionBetween/getPositionBetween");
var _degreesToRadians = require("./radians/degreesToRadians");
var _radiansToDegrees = require("./radians/radiansToDegrees");
var _clamp = require("./range/clamp/clamp");
var _convertRangePosition = require("./range/convertRangePosition/convertRangePosition");
var _getRange = require("./range/getRange/getRange");
var _isInRange = require("./range/isInRange/isInRange");
var _keepOutsideRange = require("./range/keepOutsideRange/keepOutsideRange");
var _mergeIntervals = require("./range/mergeIntervals/mergeIntervals");
var _ceilDecimals = require("./rounding/ceilDecimals/ceilDecimals");
var _ceilToMultipleOf = require("./rounding/ceilToMultipleOf/ceilToMultipleOf");
var _floorDecimals = require("./rounding/floorDecimals/floorDecimals");
var _floorToMultipleOf = require("./rounding/floorToMultipleOf/floorToMultipleOf");
var _roundDecimals = require("./rounding/roundDecimals/roundDecimals");
var _roundToMultipleOf = require("./rounding/roundToMultipleOf/roundToMultipleOf");
var _truncDecimals = require("./rounding/truncDecimals/truncDecimals");
var _isNumberRound = require("./rounding/isNumberRound/isNumberRound");
var _getAge = require("./time/getAge/getAge");
var _getDuration = require("./time/getDuration/getDuration");
var _getHalfLifeDecay = require("./time/getHalfLifeDecay/getHalfLifeDecay");
var _getTimeDifference = require("./time/getTimeDifference/getTimeDifference");
