// @baggie/core/lib/module/math/average/getFrequency/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { InputWrapper } from "@baggie/react";
import { getFrequency } from "./getFrequency";
const printMap = map => {
  return "Map {\n" + Array.from(map).reduce((obj, _ref) => {
    let [key, value] = _ref;
    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
    obj += `    ${typeof key === "string" ? `"${key}"` : JSON.parse(key)} => ${value},\n`;
    return obj;
  }, "") + "}";
};
export const Example = props => {
  const [inputString, setInputString] = useState(props.defaultInput.join(", "));
  const [returnMap, setReturnMap] = useState(false);
  const {
    input,
    result
  } = useMemo(() => {
    const inputs = inputString.split(/\s*,\s*/).filter(value => value.length).map(string => string.match(/^-?[0-9]*\.?[0-9]*$/) ? parseFloat(string) : string);
    return {
      input: inputs,
      result: getFrequency(inputs, returnMap)
    };
  }, [inputString, returnMap]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Input",
    labelForId: "input-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "input-input",
    value: inputString,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setInputString(target.value);
    }
  }))), /*#__PURE__*/React.createElement("label", null, /*#__PURE__*/React.createElement("input", {
    type: "checkbox",
    checked: returnMap,
    onChange: () => setReturnMap(current => !current)
  }), " ", "Return as Map object"), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { getFrequency } from "@baggie/core";

const input = [${input.map(item => typeof item === "string" ? `"${item}"` : item).join(", ")}];

const result = getFrequency(input${returnMap ? ", true" : ""});
/*
result = ${returnMap ? printMap(result) : JSON.stringify(result, null, 4)}
*/
`
  }));
};
