// @baggie/core/lib/module/color/convert/convertStringToRgb/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { ColorInput } from "../../_components/ColorInput/ColorInput";
import { convertStringToRgb } from "./convertStringToRgb";
export const Example = props => {
  const [hex, setHex] = useState(props.hex);
  const rgb = useMemo(() => {
    try {
      return convertStringToRgb(hex);
    } catch (err) {
      return undefined;
    }
  }, [hex]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(ColorInput, {
    label: "Hex",
    value: hex,
    onChange: setHex
  })), /*#__PURE__*/React.createElement("div", {
    style: {
      marginTop: "25px"
    }
  }, /*#__PURE__*/React.createElement("table", {
    style: {
      margin: 0
    }
  }, /*#__PURE__*/React.createElement("tbody", null, /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement("b", null, "Red:")), /*#__PURE__*/React.createElement("td", null, rgb ? rgb.red : "")), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement("b", null, "Green:")), /*#__PURE__*/React.createElement("td", null, rgb ? rgb.green : "")), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement("b", null, "Blue:")), /*#__PURE__*/React.createElement("td", null, rgb ? rgb.blue : "")), /*#__PURE__*/React.createElement("tr", null, /*#__PURE__*/React.createElement("td", null, /*#__PURE__*/React.createElement("b", null, "Alpha:")), /*#__PURE__*/React.createElement("td", null, rgb ? rgb.alpha : ""))))), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { convertStringToRgb } from "@baggie/core";

const hex = ${hex ? `"${hex}"` : "undefined"};

const rgb = convertStringToRgb(hex);
/*
rgb = ${JSON.stringify(rgb, null, 4)}
*/
`
  }));
};
