// @baggie/core/lib/commonjs/string/manipulation/substringHtmlEntitySafe/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Example = void 0;
var _react = _interopRequireWildcard(require("react"));
var _blocks = require("@storybook/blocks");
var _react2 = require("@baggie/react");
var _substringHtmlEntitySafe = require("./substringHtmlEntitySafe");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Example = props => {
  const [input, setInput] = (0, _react.useState)(props.defaultInput);
  const [start, setStart] = (0, _react.useState)(props.start);
  const [end, setEnd] = (0, _react.useState)(props.end);
  const [verbose, setVerbose] = (0, _react.useState)(false);
  const {
    output,
    realIndexStart,
    realIndexEnd
  } = (0, _react.useMemo)(() => (0, _substringHtmlEntitySafe.substringHtmlEntitySafe)(input, start, end, {
    verbose: true
  }), [input, start, end]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      flexDirection: "column",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Input",
    labelForId: "input-input"
  }, /*#__PURE__*/_react.default.createElement(_react2.TextareaWithHighlights, {
    id: "input-input",
    value: input,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setInput(target.value);
    },
    style: {
      width: "100%",
      maxWidth: "360px"
    },
    highlights: [realIndexStart, realIndexEnd],
    spellCheck: false,
    rows: 3
  })), /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Index start",
    labelForId: "start-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "start-input",
    type: "number",
    value: start,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setStart(target.valueAsNumber);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Index end",
    labelForId: "end-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "end-input",
    type: "number",
    value: end,
    onChange: _ref3 => {
      let {
        target
      } = _ref3;
      return setEnd(target.valueAsNumber);
    }
  }))), /*#__PURE__*/_react.default.createElement("label", null, /*#__PURE__*/_react.default.createElement("input", {
    type: "checkbox",
    checked: verbose,
    onChange: () => setVerbose(current => !current)
  }), "Verbose")), /*#__PURE__*/_react.default.createElement(_blocks.Source, {
    dark: true,
    code: `
import { substringHtmlEntitySafe } from "@baggie/core";

const input = \`${input.replace(/`/g, "\\`")}\`;

const indexStart = ${start || 0};
const indexEnd = ${isNaN(end) ? "undefined" : end};

const output = substringHtmlEntitySafe(input, indexStart, indexEnd${verbose ? ", { verbose: true }" : ""});
/*
${verbose ? `output = ${JSON.stringify({
      output,
      realIndexStart,
      realIndexEnd
    }, null, 4)}` : `output = \`${output}\``}
*/
`
  }));
};
exports.Example = Example;
