// @baggie/core/lib/module/string/regex/getNewlineRegex/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { InputWrapper } from "@baggie/react";
import { getNewlineRegex } from "./getNewlineRegex";
export const Example = props => {
  const [string, setString] = useState(props.string);
  const [includeWhitespace, setIncludeWhitespace] = useState(false);
  const newString = useMemo(() => string.replace(getNewlineRegex(includeWhitespace), "-"), [string, includeWhitespace]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(InputWrapper, {
    label: "String",
    labelForId: "string-input"
  }, /*#__PURE__*/React.createElement("textarea", {
    id: "string-input",
    value: string,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setString(target.value);
    },
    style: {
      width: "100%",
      maxWidth: "360px"
    },
    spellCheck: false,
    rows: 3
  })), /*#__PURE__*/React.createElement("label", null, /*#__PURE__*/React.createElement("input", {
    type: "checkbox",
    checked: includeWhitespace,
    onChange: () => setIncludeWhitespace(x => !x)
  }), " ", "Include whitespace"), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { getNewlineRegex } from "@baggie/core";

const string = \`${string}\`;

const newString = string.replace(getNewlineRegex(${includeWhitespace ? "true" : ""}), "-");
// newString = \`${newString}\`
`
  }), /*#__PURE__*/React.createElement("b", null, "The returned regex:"), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `new RegExp("${getNewlineRegex(includeWhitespace).source}", "${getNewlineRegex(includeWhitespace).flags}")`
  }));
};
