// @baggie/core/lib/commonjs/math/rounding/isNumberRound/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Example = void 0;
var _react = _interopRequireWildcard(require("react"));
var _blocks = require("@storybook/blocks");
var _react2 = require("@baggie/react");
var _isNumberRound = require("./isNumberRound");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Example = props => {
  const [number, setNumber] = (0, _react.useState)(props.number);
  const [maxDecimalPlaces, setMaxDecimalPlaces] = (0, _react.useState)(props.maxDecimalPlaces);
  const result = (0, _react.useMemo)(() => (0, _isNumberRound.isNumberRound)(number, maxDecimalPlaces), [number, maxDecimalPlaces]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Number",
    labelForId: "number-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "number-input",
    type: "number",
    value: number,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setNumber(target.valueAsNumber);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Max decimal places",
    labelForId: "max-decimal-places-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "max-decimal-places-input",
    type: "number",
    value: maxDecimalPlaces,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setMaxDecimalPlaces(target.valueAsNumber);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Result",
    labelForId: undefined
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, result ? "true" : "false"))), /*#__PURE__*/_react.default.createElement(_blocks.Source, {
    dark: true,
    code: `
import { isNumberRound } from "@baggie/core";

const number = ${number};
const maxDecimalPlaces = ${maxDecimalPlaces};

const result = isNumberRound(number, maxDecimalPlaces);
// result = ${result ? "true" : "false"}
`
  }));
};
exports.Example = Example;
