// @baggie/core/lib/module/math/geometry/circle/getCircleCircumference/Example.js
import React, { useMemo, useState } from "react";
import { InputWrapper } from "@baggie/react";
import { Source } from "@storybook/blocks";
import { getCircleCircumference } from "./getCircleCircumference";
export const Example = props => {
  const [radius, setRadius] = useState(props.radius);
  const circumference = useMemo(() => getCircleCircumference(radius || 0), [radius]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Radius",
    labelForId: "radius-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "radius-input",
    type: "number",
    value: radius,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setRadius(target.valueAsNumber);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Circumference",
    labelForId: undefined
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, isNaN(circumference) ? "Error" : circumference))), /*#__PURE__*/React.createElement("div", {
    style: {
      aspectRatio: 1,
      background: "#FC0",
      marginTop: "20px",
      borderRadius: "50%",
      position: "relative",
      display: "inline-grid",
      gridTemplateColumns: "1fr 1fr",
      alignItems: "center",
      pointerEvents: "none",
      userSelect: "none"
    }
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      position: "absolute",
      inset: "-4px",
      borderRadius: "50%",
      border: "2px dashed #900"
    }
  }), /*#__PURE__*/React.createElement("div", null), /*#__PURE__*/React.createElement("div", {
    style: {
      top: "-0.75em",
      padding: "0 10px",
      position: "relative",
      borderBottom: "1px solid #900"
    }
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      lineHeight: "1.5em"
    }
  }, "r = ", radius || 0))), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { getCircleCircumference } from "@baggie/core";

const radius = ${radius || 0};

const circumference = getCircleCircumference(radius);
// circumference = ${circumference}
`
  }));
};
