// @baggie/core/lib/module/math/rounding/isNumberRound/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { InputWrapper } from "@baggie/react";
import { isNumberRound } from "./isNumberRound";
export const Example = props => {
  const [number, setNumber] = useState(props.number);
  const [maxDecimalPlaces, setMaxDecimalPlaces] = useState(props.maxDecimalPlaces);
  const result = useMemo(() => isNumberRound(number, maxDecimalPlaces), [number, maxDecimalPlaces]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Number",
    labelForId: "number-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "number-input",
    type: "number",
    value: number,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setNumber(target.valueAsNumber);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Max decimal places",
    labelForId: "max-decimal-places-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "max-decimal-places-input",
    type: "number",
    value: maxDecimalPlaces,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setMaxDecimalPlaces(target.valueAsNumber);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Result",
    labelForId: undefined
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, result ? "true" : "false"))), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { isNumberRound } from "@baggie/core";

const number = ${number};
const maxDecimalPlaces = ${maxDecimalPlaces};

const result = isNumberRound(number, maxDecimalPlaces);
// result = ${result ? "true" : "false"}
`
  }));
};
