// @baggie/core/lib/commonjs/math/range/clamp/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Example = void 0;
var _react = _interopRequireWildcard(require("react"));
var _blocks = require("@storybook/blocks");
var _react2 = require("@baggie/react");
var _clamp = require("./clamp");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Example = props => {
  const [input, setInput] = (0, _react.useState)(props.input);
  const [min, setMin] = (0, _react.useState)(props.min);
  const [max, setMax] = (0, _react.useState)(props.max);
  const result = (0, _react.useMemo)(() => (0, _clamp.clamp)(input, {
    min,
    max
  }), [input, min, max]);
  const settings = [typeof min === "number" ? `min: ${min}` : "", typeof max === "number" ? `max: ${max}` : ""].filter(x => x).join(", ");
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Input",
    labelForId: "input-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "input-input",
    type: "number",
    value: input,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setInput(target.valueAsNumber || 0);
    },
    autoComplete: "off"
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Min",
    labelForId: "min-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "min-input",
    type: "number",
    value: min,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setMin(isNaN(target.valueAsNumber) ? undefined : target.valueAsNumber);
    },
    autoComplete: "off"
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Max",
    labelForId: "max-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "max-input",
    type: "number",
    value: max,
    onChange: _ref3 => {
      let {
        target
      } = _ref3;
      return setMax(isNaN(target.valueAsNumber) ? undefined : target.valueAsNumber);
    },
    autoComplete: "off"
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Result",
    labelForId: undefined
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, result))), /*#__PURE__*/_react.default.createElement(_blocks.Source, {
    dark: true,
    code: `
import { clamp } from "@baggie/core";

const result = clamp(${input}, { ${settings} });
// result = ${result}
`
  }));
};
exports.Example = Example;
