// @baggie/core/lib/commonjs/math/rounding/ceilToMultipleOf/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Example = void 0;
var _react = _interopRequireWildcard(require("react"));
var _blocks = require("@storybook/blocks");
var _react2 = require("@baggie/react");
var _ceilToMultipleOf = require("./ceilToMultipleOf");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Example = props => {
  const [number, setNumber] = (0, _react.useState)(props.number);
  const [multipleOf, setMultipleOf] = (0, _react.useState)(props.multipleOf);
  const result = (0, _react.useMemo)(() => (0, _ceilToMultipleOf.ceilToMultipleOf)(number, multipleOf), [number, multipleOf]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Number",
    labelForId: "number-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "number-input",
    type: "number",
    value: number,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setNumber(target.valueAsNumber);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Multiple of",
    labelForId: "multiple-of-input"
  }, /*#__PURE__*/_react.default.createElement("input", {
    id: "multiple-of-input",
    type: "number",
    value: multipleOf,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setMultipleOf(target.valueAsNumber);
    }
  })), /*#__PURE__*/_react.default.createElement(_react2.InputWrapper, {
    label: "Result",
    labelForId: undefined
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, isNaN(result) ? "Error" : result))), /*#__PURE__*/_react.default.createElement(_blocks.Source, {
    dark: true,
    code: `
import { ceilToMultipleOf } from "@baggie/core";

const number = ${number};
const multipleOf = ${multipleOf};

const result = ceilToMultipleOf(number, multipleOf);
// result = ${result}
`
  }));
};
exports.Example = Example;
