// @baggie/core/lib/module/math/time/getHalfLifeDecay/Example.js
import React, { useMemo, useState } from "react";
import { InputWrapper } from "@baggie/react";
import { Source } from "@storybook/blocks";
import { getHalfLifeDecay } from "./getHalfLifeDecay";
export const Example = props => {
  const [initialValue, setInitialValue] = useState(props.initialValue);
  const [timePassed, setTimePassed] = useState(props.timePassed);
  const [halfLife, setHalfLife] = useState(props.halfLife);
  const reducedValue = useMemo(() => getHalfLifeDecay(initialValue, timePassed || 0, halfLife || 0), [initialValue, timePassed, halfLife]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Initial value",
    labelForId: "value-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "value-input",
    type: "number",
    value: initialValue,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setInitialValue(target.valueAsNumber);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Time passed",
    labelForId: "time-passed-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "time-passed-input",
    type: "number",
    value: timePassed,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setTimePassed(target.valueAsNumber);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Half life",
    labelForId: "half-life-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "half-life-input",
    type: "number",
    value: halfLife,
    onChange: _ref3 => {
      let {
        target
      } = _ref3;
      return setHalfLife(target.valueAsNumber);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Reduced value",
    labelForId: undefined
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, isNaN(reducedValue) ? "Error" : reducedValue))), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { getHalfLifeDecay } from "@baggie/core";

const initialValue = ${initialValue || 0};
const timePassed = ${timePassed || 0};
const halfLife = ${halfLife || 0};

const reducedValue = getHalfLifeDecay(initialValue, timePassed, halfLife);
// reducedValue = ${reducedValue};
`
  }));
};
