// @baggie/core/lib/module/color/convert/convertColorNameToHex/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { InputWrapper } from "@baggie/react";
import { ColorViewer } from "../../_components/ColorViewer/ColorViewer";
import { COLOR_NAMES, convertColorNameToHex } from "./convertColorNameToHex";
export const Example = props => {
  const [colorName, setColorName] = useState(props.colorName);
  const hex = useMemo(() => convertColorNameToHex(colorName), [colorName]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Color name",
    labelForId: "color-name-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "color-name-input",
    value: colorName,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setColorName(target.value);
    },
    list: "color-names"
  }), /*#__PURE__*/React.createElement("datalist", {
    id: "color-names"
  }, Object.keys(COLOR_NAMES).map(name => /*#__PURE__*/React.createElement("option", {
    key: name
  }, name))))), /*#__PURE__*/React.createElement("div", {
    style: {
      marginTop: "25px"
    }
  }, /*#__PURE__*/React.createElement(ColorViewer, {
    color: hex
  })), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { convertColorNameToHex } from "@baggie/core";

const colorName = "${colorName}";

const hex = convertColorNameToHex(colorName);
// hex = ${hex ? `"${hex}"` : "undefined"}
`
  }));
};
