// @baggie/core/lib/module/math/arithmetic/sortByDifference/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { InputWrapper } from "@baggie/react";
import { sortByDifference } from "./sortByDifference";
export const Example = props => {
  const [target, setTarget] = useState(props.target);
  const [inputNumbers, setInputNumbers] = useState(props.numbers.join(", "));
  const {
    numbers,
    sorted
  } = useMemo(() => {
    const numbers = inputNumbers.split(/\s*,\s*/).filter(value => value.length).map(string => string.match(/^-?[0-9]*\.?[0-9]*$/) ? parseFloat(string) : NaN);
    return {
      numbers,
      sorted: sortByDifference(target, numbers)
    };
  }, [target, inputNumbers]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Target",
    labelForId: "target-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "target-input",
    type: "number",
    value: target,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setTarget(target.valueAsNumber);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Numbers",
    labelForId: "numbers-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "numbers-input",
    value: inputNumbers,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setInputNumbers(target.value);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Sorted",
    labelForId: undefined
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, "[", sorted.join(", "), "]"))), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { sortByDifference } from "@baggie/core";

const target = ${target};
const numbers = [${numbers.join(", ")}];

const sorted = sortByDifference(target, numbers);
// sorted = [${sorted.join(", ")}]
`
  }));
};
