// @baggie/core/lib/module/string/regex/getVimeoIdRegex/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { InputWrapper } from "@baggie/react";
import { getVimeoIdRegex } from "./getVimeoIdRegex";
export const Example = props => {
  const [string, setString] = useState(props.string);
  const match = useMemo(() => string.match(getVimeoIdRegex()), [string]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(InputWrapper, {
    label: "String",
    labelForId: "string-input"
  }, /*#__PURE__*/React.createElement("textarea", {
    id: "string-input",
    value: string,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setString(target.value);
    },
    style: {
      width: "100%",
      maxWidth: "360px"
    },
    spellCheck: false,
    rows: 3
  })), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { getVimeoIdRegex } from "@baggie/core";

const string = "${string}";

const matchVimeoId = string.match(getVimeoIdRegex());
// matchVimeoId = ${JSON.stringify(match)}
`
  }), /*#__PURE__*/React.createElement("b", null, "The returned regex:"), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `new RegExp("${getVimeoIdRegex().source}", "${getVimeoIdRegex().flags}")`
  }));
};
