// @baggie/core/lib/module/string/manipulation/convertNewlinesToBreaks/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { InputWrapper } from "@baggie/react";
import { convertNewlinesToBreaks } from "./convertNewlinesToBreaks";
export const Example = props => {
  const [input, setInput] = useState(props.input);
  const [keepWhitespace, setKeepWhitespace] = useState(false);
  const output = useMemo(() => convertNewlinesToBreaks(input, keepWhitespace), [input, keepWhitespace]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      flexDirection: "column",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Input",
    labelForId: "input-input"
  }, /*#__PURE__*/React.createElement("textarea", {
    id: "input-input",
    value: input,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setInput(target.value);
    },
    style: {
      width: "360px",
      maxWidth: "100%"
    },
    spellCheck: false,
    rows: 3
  })), /*#__PURE__*/React.createElement("label", null, /*#__PURE__*/React.createElement("input", {
    type: "checkbox",
    checked: keepWhitespace,
    onChange: () => setKeepWhitespace(current => !current)
  }), "Keep whitespace")), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { convertNewlinesToBreaks } from "@baggie/core";

const string = \`${input}\`;
${keepWhitespace ? "const keepWhitespace = true;\n" : ""}
const output = convertNewlinesToBreaks(string${keepWhitespace ? ", keepWhitespace" : ""});
// output = ${output}
`
  }));
};
