// @baggie/core/lib/module/math/average/getAntiMode/Example.js
import React, { useMemo, useState } from "react";
import { Source } from "@storybook/blocks";
import { InputWrapper } from "@baggie/react";
import { getAntiMode } from "./getAntiMode";
export const Example = props => {
  const [input, setInput] = useState(props.defaultInput.join(", "));
  const {
    numbers,
    result
  } = useMemo(() => {
    const numbers = input.split(/\s*,\s*/).filter(value => value.length).map(string => string.match(/^-?[0-9]*\.?[0-9]*$/) ? parseFloat(string) : NaN);
    return {
      numbers,
      result: getAntiMode(numbers)
    };
  }, [input]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Input",
    labelForId: "input-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "input-input",
    value: input,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setInput(target.value);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Output",
    labelForId: undefined
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, result ? `[${result.numbers.join(", ")}]` : "Error"))), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { getAntiMode } from "@baggie/core";

const numbers = [${numbers.join(", ")}];

const result = getAntiMode(numbers);
/*
result = ${JSON.stringify(result, null, 4)}
*/
`
  }));
};
