// @baggie/core/lib/commonjs/color/convert/convertStringToRgb/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Example = void 0;
var _react = _interopRequireWildcard(require("react"));
var _blocks = require("@storybook/blocks");
var _ColorInput = require("../../_components/ColorInput/ColorInput");
var _convertStringToRgb = require("./convertStringToRgb");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Example = props => {
  const [hex, setHex] = (0, _react.useState)(props.hex);
  const rgb = (0, _react.useMemo)(() => {
    try {
      return (0, _convertStringToRgb.convertStringToRgb)(hex);
    } catch (err) {
      return undefined;
    }
  }, [hex]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/_react.default.createElement(_ColorInput.ColorInput, {
    label: "Hex",
    value: hex,
    onChange: setHex
  })), /*#__PURE__*/_react.default.createElement("div", {
    style: {
      marginTop: "25px"
    }
  }, /*#__PURE__*/_react.default.createElement("table", {
    style: {
      margin: 0
    }
  }, /*#__PURE__*/_react.default.createElement("tbody", null, /*#__PURE__*/_react.default.createElement("tr", null, /*#__PURE__*/_react.default.createElement("td", null, /*#__PURE__*/_react.default.createElement("b", null, "Red:")), /*#__PURE__*/_react.default.createElement("td", null, rgb ? rgb.red : "")), /*#__PURE__*/_react.default.createElement("tr", null, /*#__PURE__*/_react.default.createElement("td", null, /*#__PURE__*/_react.default.createElement("b", null, "Green:")), /*#__PURE__*/_react.default.createElement("td", null, rgb ? rgb.green : "")), /*#__PURE__*/_react.default.createElement("tr", null, /*#__PURE__*/_react.default.createElement("td", null, /*#__PURE__*/_react.default.createElement("b", null, "Blue:")), /*#__PURE__*/_react.default.createElement("td", null, rgb ? rgb.blue : "")), /*#__PURE__*/_react.default.createElement("tr", null, /*#__PURE__*/_react.default.createElement("td", null, /*#__PURE__*/_react.default.createElement("b", null, "Alpha:")), /*#__PURE__*/_react.default.createElement("td", null, rgb ? rgb.alpha : ""))))), /*#__PURE__*/_react.default.createElement(_blocks.Source, {
    dark: true,
    code: `
import { convertStringToRgb } from "@baggie/core";

const hex = ${hex ? `"${hex}"` : "undefined"};

const rgb = convertStringToRgb(hex);
/*
rgb = ${JSON.stringify(rgb, null, 4)}
*/
`
  }));
};
exports.Example = Example;
