// @baggie/core/lib/module/math/geometry/rectangle/getRectanglePerimeter/Example.js
import React, { useMemo, useState } from "react";
import { InputWrapper } from "@baggie/react";
import { Source } from "@storybook/blocks";
import { getRectanglePerimeter } from "./getRectanglePerimeter";
export const Example = props => {
  const [x, setX] = useState(props.x);
  const [y, setY] = useState(props.y);
  const perimeter = useMemo(() => getRectanglePerimeter(x, y), [x, y]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex",
      gap: "10px"
    }
  }, /*#__PURE__*/React.createElement(InputWrapper, {
    label: "X",
    labelForId: "x-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "x-input",
    type: "number",
    value: x,
    onChange: _ref => {
      let {
        target
      } = _ref;
      return setX(target.valueAsNumber);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Y",
    labelForId: "x-input"
  }, /*#__PURE__*/React.createElement("input", {
    id: "y-input",
    type: "number",
    value: y,
    onChange: _ref2 => {
      let {
        target
      } = _ref2;
      return setY(target.valueAsNumber);
    }
  })), /*#__PURE__*/React.createElement(InputWrapper, {
    label: "Perimeter",
    labelForId: undefined
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      padding: "5px 0"
    }
  }, isNaN(perimeter) ? "Error" : perimeter))), /*#__PURE__*/React.createElement("div", {
    style: {
      aspectRatio: `${Math.abs(x)} / ${Math.abs(y)}`,
      background: "#FC0",
      width: Math.abs(x) >= Math.abs(y) ? "150px" : `${Math.abs(x) / Math.abs(y) * 150}px`,
      height: Math.abs(x) <= Math.abs(y) ? "150px" : `${Math.abs(y) / Math.abs(x) * 150}px`,
      marginTop: "20px",
      position: "relative",
      display: "inline-block",
      pointerEvents: "none",
      userSelect: "none",
      transition: "all .4s ease-out"
    }
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      position: "absolute",
      inset: "-4px",
      border: "2px dashed #900"
    }
  })), /*#__PURE__*/React.createElement(Source, {
    dark: true,
    code: `
import { getRectanglePerimeter } from "@baggie/core";

const x = ${x || 0};
const y = ${y || 0};

const perimeter = getRectanglePerimeter(x, y);
// perimeter = ${perimeter}
`
  }));
};
